mycore-base/src/main/java/org/mycore/access/strategies/MCRCreatorRuleStrategy.java

```diff
@@ -118,24 +147,24 @@ public boolean hasRuleMapping(String id, String permission) {
     }
 
     public boolean isCreatorRuleAvailable(String id, String permission) {
-        if (CREATOR_PERMISSIONS.contains(permission)) {
-            MCRObjectID mcrObjectId = null;
-            try {
-                mcrObjectId = MCRObjectID.getInstance(id);
-                MCRUserInformation currentUser = MCRSessionMgr.getCurrentSession().getUserInformation();
-                if (CREATOR_ROLES.stream().anyMatch(currentUser::isUserInRole)
-                    && objectStatusIsSubmitted(mcrObjectId)) {
-                    if (isCurrentUserCreator(mcrObjectId, currentUser)) {
-                        return true;
-                    }
+        MCRObjectID mcrObjectId = null;
+        try {
+            mcrObjectId = MCRObjectID.getInstance(id);
+            MCRUserInformation currentUser = MCRSessionMgr.getCurrentSession().getUserInformation();
+            if (userHasCreatorRole(currentUser) && userIsCreatorOf(currentUser, mcrObjectId)) {
+                if (objectStatusIsSubmitted(mcrObjectId) && CREATOR_SUBMITTED_PERMISSIONS.contains(permission)) {
```

Comment:
Should be changed to:
```
if (objectStatusIsSubmitted(mcrObjectId)){
return CREATOR_SUBMITTED_PERMISSIONS.contains(permission);
} else if(objectStatusIsReview(mcrObjectId)) {
return CREATOR_REVIEW_PERMISSIONS.contains(permission);
}
```
to prevent double status check

Reply:
The code, as proposed, is a more readable version of

```
 if (objectStatusIsSubmitted(mcrObjectId) && CREATOR_SUBMITTED_PERMISSIONS.contains(permission) || (objectStatusIsReview(mcrObjectId) && CREATOR_REVIEW_PERMISSIONS.contains(permission)) {
   return true;
 }
```

I intentionally avoided the use of `else` here, because the two branches are equally significant. The proposed change obfuscates this intent and introduces a subtle difference.

Assume the following configuration:

```
MCR.Access.Strategy.SubmittedCategories         = s1,s2
MCR.Access.Strategy.ReviewCategories            = s2,s3
MCR.Access.Strategy.CreatorSubmittedPermissions = p1,p2
MCR.Access.Strategy.CreatorReviewPermissions    = p1,p3
```

Now, if the object is in state `s2` and the requested permission is `p3`, I' expect a result of `true`, but with the proposed change, the actual result would be `false`.

I've no idea if such a configuration is in any way realistic, but on the other hand, I see no reason to disallow such a configuration.

Should someone have a need for such a configuration, it would be very hard for them to detect that is is not supported. Best case would than probably a lengthy debugging session; worst case would be some unexpected and unnoticed permission-problem. I think we shouldn't unnecessarily introduce such subtleties.